lib/syncthing/syncthing.go

```diff
@@ -376,6 +391,17 @@ func (a *App) run() {
 	close(a.stopped)
 }
 
+func (a *App) EventLogger() (*events.Logger, error) {
```

Comment:
GetEventLogger?
Also, feels like something that should be injected from cmd, oppose to retrievable from the lib, then app should not even be responsible for running it, which then could be injected into the autoupgrade path.

Reply:
Sure, definitely the straight forward thing to do. An argument for retrieving is that some other consumer of `lib/syncthing`, which might not care at all about events (e.g. Martchus' syncthingtray), does not have to deal with it, but is forced to when injected. Of course it isn't much that they'd need to do (define the event logger type, call yet another utility function and pass the object to the app creator). What do you think about that - should I still change to injection?

And just in case: We are pretty inconsistent with using the `Get` prefix (or not using it) - I could easily find a few examples for both. Personally I like it better without, but I don't care about it, I'll change if needed (see above).

Reply:
You still reach out into the app to grab it for upgrades, which means either upgrades need to move into lib/syncthing, or event creation needs to move out.

Re naming, not that I think about it, it's os.Hostname() not os.GetHostname(), so probably I am just biast having written a lot of Java.

Reply:
I don't understand how using events in cmd/syncthing requires upgrades to move into lib/syncthing or event creation to move out of lib/syncthing. In my view upgrades are a thing of the released Syncthing binary, nothing the library needs to know about (well it has an exit signal for upgrading because of the upgrade rest endpoint). And one criterion was that the library should ideally not interact with the FS. I am not much worried about moving events creation out of the library, my only argument against it is convenience, I'd just like to understand the need for doing so.

Reply:
For me it's a question of general principles, dependency injection / inversion of control. The purpose of lib/syncthing is not to be a creator of event systems, so it is best to leave it to the caller. That way the caller can pass an event implementation that does what they like, including one that is suitable for testing.